PlottingAnalysis/TS_TopFeatures.m

```diff
@@ -125,10 +147,12 @@
     classNumbers = arrayfun(@(x)sum(TimeSeries.Group==x),1:numClasses);
     isBalanced = all(classNumbers==classNumbers(1));
     if isBalanced
-        fn_testStat = GiveMeFunctionHandle(whatTestStat,numClasses,'acc',0);
+        fn_testStat = @(XTrain,yTrain,XTest,yTest,numFolds) GiveMeCfn(whatTestStat,...
+                            XTrain,yTrain,XTest,yTest,numClasses,0,'acc',[],numFolds);
         fprintf(1,'Using overall classification accuracy as output measure\n');
     else
-        fn_testStat = GiveMeFunctionHandle(whatTestStat,numClasses,'balancedAcc',1);
+        fn_testStat = @(XTrain,yTrain,XTest,yTest,numFolds) GiveMeCfn(whatTestStat,...
```

Comment:
Same with this one… Isn't the point of `GiveMeFunctionHandle` to save this kind of syntax?
Or perhaps there is a case for having just a single function rather than both `GiveMeFunctionHandle` and `GiveMeCfn`...

Reply:
Looks like I've missed some crucial code in the commit, `GiveMeCfn` should have the following code in it:

```
if strcmp(whatClassifier,'fast_linear')
    if CVFolds > 0
        Mdl = fitcdiscr(XTrain,yTrain,'Prior','uniform','KFold',CVFolds);
    else
        Mdl = fitcdiscr(XTrain,yTrain,'Prior','uniform');
    end
else
```
 
And yep, I think with this addition that we won't need the two functions anymore, since `fitcdiscr` with the uniform prior is (supposedly) just as fast as classify

Reply:
Even with the current form (checking for binary first for speed), it would be cleaner in a single function…
Can you add a commit to this PR for the code you mention above?

Reply:
Reluctant to give up `classify` until (at least quickly) testing the new code is as fast…

Reply:
I've just tested the two on the `fisheriris` data and turns out that running `fitcdiscr(XTrain,yTrain,'Prior','uniform')` followed by `predict(...)` is 2x slower than just running `classify`.

Will add the code snippet in a new commit

Reply:
Awesome—thanks for testing! This would all be much nicer in scikit learn…

Reply:
I've also added in that we use `classify` if only the accuracy (first output) is required and switch to `fitcdiscr` otherwise. So we maintain that efficiency